src/Chapter1.hs

```diff
@@ -653,7 +669,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n = let x = floor (fraction / 10) in if x < 10 then x else firstDigit x
+    where 
+        fraction = fromIntegral n :: Float
```

Comment:
Wow, this is pretty interesting solution 👏🏼 
Though, you can get rid of the to/from Float conversion if you use `div` function 🙂 As you are only dealing with `Int`s in this task it is totally fine 👌🏼